client/components/search/index.jsx

```diff
@@ -266,10 +267,13 @@ const Search = React.createClass( {
 				this.props.initialValue;
 
 		const autocorrect = this.props.disableAutocorrect && {
-			autoComplete: 'off',
-			autoCorrect: 'off',
-			spellCheck: 'false'
-		};
+				autoComplete: 'off',
+				autoCorrect: 'off',
+				spellCheck: 'false'
+			},
+			maxLength = this.props.maxLength && {
```

Comment:
I think this is a bit of an overkill and makes it less readable. A simple `maxLength = this.props.maxLength` here and `maxLength = { maxLength }` on the component should suffice. You were probably basing this on the `autocorrect` variable - but that one has more than one property, so it makes sense there :)